x-pack/plugins/observability_solution/apm/server/lib/connections/get_connection_stats/get_destination_map.ts

```diff
@@ -67,52 +70,78 @@ export const getDestinationMap = ({
       offset,
     });
 
+    const query: QueryDslQueryContainer = {
+      bool: {
+        filter: [
+          { exists: { field: SPAN_DESTINATION_SERVICE_RESOURCE } },
+          ...rangeQuery(startWithOffset, endWithOffset),
+          ...filter,
+          ...excludeRumExitSpansQuery(),
+        ],
+      },
+    };
+
+    const hitCountResponse = await apmEventClient.search('get_exit_span_doc_count', {
+      apm: {
+        events: [ProcessorEvent.span],
+      },
+      body: {
+        size: 0,
+        track_total_hits: true,
+        query,
+      },
+    });
+
+    const totalDocCount = hitCountResponse.hits.total.value;
+    const rawSamplingProbability = Math.min(100_000 / totalDocCount, 1);
+    const samplingProbability =
+      rawSamplingProbability < 0.5 ? rawSamplingProbability : randomSampler.probability;
```

Comment:
It seems like `randomSampler.probability` is always going to be 1, and we want it to be 1, so I wonder why use this property.

Reply:
`randomSampler.probability` will be `1` if `1` is passed to `getRandomSampler`. We could hardcode `1` here, but if we keep consistency with other places in APM, we will want to use `randomSampler.probability`. 

The difference between here and other places in APM is that in some [routes](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/routes/storage_explorer/route.ts#L60), is that the `probability` can be passed in the request - which could be implemented in the future for this function.

Reply:
Since we're using the randomSampler object for the `seed` as well, I personally don't see too much of an issue with using `randomSampler.probability` when the sampler rate is > than 0.5. 

Even if `1` is fixed in the random ampler agg configuration, [like here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/routes/assistant_functions/get_observability_alert_details_context/index.ts#L78)

It would be fine to hardcode 1 here if we weren't using `getRandomSampler` to build the configuration though. wdyt?

Reply:
> Since we're using the randomSampler object for the seed as well, I personally don't see too much of an issue with using randomSampler.probability when the sampler rate is > than 0.5.

But we don't let the user ever configure the seed right?  And all the seed calculations are done within the function that returns it?

I think part of the problem is wanting to colocate the probability with the seed when this function was created but not supporting passing in your own function to the probability.  So now our probability calculation is not colocated with the probability value required in the getRandomSampler function, unlike the seed which is.

I'm not too fussed about it, just think this makes it harder to read with unexpected behaviour if we allow the value to be passed into the route as you mentioned.  The user would have to be aware of the conditions under which its used/not used.

Reply:
> But we don't let the user ever configure the seed right? And all the seed calculations are done within the function that returns it?

the seed helps the query utilize the same subset of documents between calls. Users are not supposed to configure that, 

> I think part of the problem is wanting to colocate the probability with the seed when this function was created but not supporting passing in your own function to the probability. So now our probability calculation is not colocated with the probability value required in the getRandomSampler function, unlike the seed which is.

That's true. The way I see it, it is more about where to configure the default probability. It's either hardcode here, or in `getRandomSampler`.

If we use the seed from `getRandomSampler`, but hardcode `1` in the `probability` it would also be a bit confusing, because the `probability` parameter in `getRandomSampler`  is mandatory. So we'd have 2 places hardcoding the default `probability. Or we hardcode both `seed` and `probability` here. Idk.